WeeWX 4 Python 2 and 3 Changes.

chaunceygardiner opened the pull request:
I’ve been successfully using the forecast plugin in WeeWX 4 for some time now.  I’ve mentioned it on weewx-development and at least one other has used it with success.  That person also asked that I send a pull request.  So, I’m sending this just in case you are interested.

These changes require WeeWX 4 and work with Python 2 and 3.

chaunceygardiner committed:
Fix py3 errors.

chaunceygardiner committed:
Working with WeeWX 4.x and Py3.

chaunceygardiner committed:
BugFix: parse parameter needed to be named.

chaunceygardiner committed:
Enforce WeeWX 4 requirement.

chaunceygardiner committed:
Fix XTideForecast

chaunceygardiner committed:
UKMO now requires User-Agent to be specified.

chaunceygardiner committed:
Merge branch 'master' of github.com:chaunceygardiner/weewx-forecast

chaunceygardiner committed:
Update readme to install zip file rather than tgz (as downloading from github yields a zip file).

chaunceygardiner committed:
Add support for testing Aeris download at the command line.

chaunceygardiner committed:
Get the name of the zip file correct.

chaunceygardiner committed:
Bump version to 3.4.0b2.

chaunceygardiner committed:
Use new style manager/schema/table_name/database spec in weewx.conf.  Use new weewx.manager.get_manager_dict_from_config.

peters77 commented:
I sucessfully used the forecast-extension (version/branch from @chaunceygardiner ) with Python 3 and weewx 4.1.1, but now I got this if I set Debug=1 :
`Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float'
Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float'`

Any idea howto fix that...or can I just ignore it?! 

Regards,

Christian

chaunceygardiner commented:
I threw in some stack traces and I see the additional places in four functions that I need to check for None.

Today is a busy work [from home] day, but I will push out a new version of chaunceygardiner/weewx-forecast tonight.

> On Jun 22, 2020, at 6:15 AM, peters77 <notifications@github.com> wrote:
> 
> ﻿
> I sucessfully used the forecast-extension (version/branch from @chaunceygardiner ) with Python 3 and weewx 4.1.1, but now I got this if I set Debug=1 :
> Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float'
> 
> Any idea howto fix that...or can I just ignore it?!
> 
> Regards,
> 
> Christian
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

chaunceygardiner committed:
Avoid logdbg output where a proposed value passed to max was None.  Add stack traces for value error (at least for now).

chaunceygardiner commented:
Hi Christian,

It turns out the _get_stats messages were benign; but I’ve cleared them up.

I can’t reproduce the get_min and get_max errors but I’ve added stack traces so I can track them down.  Would you please pull the new version and report back with the output you now get when setting debug=1?

Cheers,
John

> On Jun 22, 2020, at 7:12 AM, John Kline <john@johnkline.com> wrote:
> 
> ﻿
> I threw in some stack traces and I see the additional places in four functions that I need to check for None.
> 
> Today is a busy work [from home] day, but I will push out a new version of chaunceygardiner/weewx-forecast tonight.
> 
>>> On Jun 22, 2020, at 6:15 AM, peters77 <notifications@github.com> wrote:
>>> 
>> ﻿
>> I sucessfully used the forecast-extension (version/branch from @chaunceygardiner ) with Python 3 and weewx 4.1.1, but now I got this if I set Debug=1 :
>> Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_stats: '>' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_min: '<' not supported between instances of 'NoneType' and 'float' Jun 22 15:00:16 weewx weewx[11654] DEBUG user.forecast: ReportThread: _get_max: '>' not supported between instances of 'NoneType' and 'float'
>> 
>> Any idea howto fix that...or can I just ignore it?!
>> 
>> Regards,
>> 
>> Christian
>> 
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub, or unsubscribe.

peters77 commented:
Hi John,

just tested with OVM, debug messages gone, log now cleared! 
Looks good! :-) 

Thank you very much for you efforts! 

Regards,

Christian

chaunceygardiner committed:
Move forecast db insertions to main thread to avoid intermittent error after weewx/weewx commit b6d2424482c2a7dc6b46e7f542c644e13383baea

chaunceygardiner committed:
Revert "Move forecast db insertions to main thread to avoid intermittent error after weewx/weewx commit b6d2424482c2a7dc6b46e7f542c644e13383baea"

This reverts commit 12d2a03fb2a5204da00d138ac3dc020b7ac15002.

This change was necessary due to a bug introduced in WeeWX when copying a configobj object.
WeeWX was never released with the bug and was fixed at:
https://github.com/weewx/weewx/commit/b6905e5ac76f2cc8128172b75854a69c0d6a9dfd

As such, this change, to move database inserts to the main thread, can now be reverted.

chaunceygardiner committed:
Update to 3.5.0b5 for release.

chaunceygardiner committed:
Update readme to reflect current version: v3.4.0b5.

chaunceygardiner committed:
Add support for the new Weather Underground API.

chaunceygardiner committed:
Handle NWS's 'MM' for missing data in more places.

chaunceygardiner committed:
forecast_iconic nad forecast_compact show_XXX bug fix.

chaunceygardiner committed:
forecat_compact now uses forecast_compact_settings.

chaunceygardiner committed:
WU: Derive cloud icon from cloud cover percentage.

chaunceygardiner committed:
Push 3.4.0b7.

chaunceygardiner committed:
v3.5.0b8: suport rain and snow icons in WU forecasts.

chaunceygardiner committed:
v2.4.0.b9: Add support for the following icons in WU forecasts: tstms, hail, dirzzle, sleet, frzngdrzl, frzngrain, flurries.

chaunceygardiner committed:
Improve header alignment on forecast strip.

chaunceygardiner committed:
v3.4.b10: Add missing hail icon.  For WU: Withold percip icons if pop <= 20%.

chaunceygardiner committed:
Freezing rain icon was not included in install script.

chaunceygardiner committed:
temperatureMin and temperatureMax were being parsed incorrectly for WU forecasts.

chaunceygardiner committed:
3.4.0b11 release.

chaunceygardiner committed:
One more b11 change.

chaunceygardiner committed:
Allow a user to specify lid_desc for NWS forecasts (a special case, see changelog).

chaunceygardiner committed:
Some HTML fixes.

chaunceygardiner committed:
More fixes to pass html and css validation.

chaunceygardiner committed:
Fix bytes vs. str issue resulting in stack trace in log under weewx 5.x.

chaunceygardiner committed:
Add notes on how to get xtide working (since it is missing from debian bookworm, and perhaps other distributions)

chaunceygardiner committed:
Update instructions on building tide.

chaunceygardiner committed:
Release 3.5: Documenations changes: WeeWX 5 installation instructions and instructions on how to build xtide.

chaunceygardiner committed:
More explicit instructions for building xtide.

chaunceygardiner committed:
Update version in readme.

chaunceygardiner committed:
Release v4.1.  Remove depdendency on six.  Python 2 is long dead.

chaunceygardiner committed:
Doc tweak for v4.1.

chaunceygardiner committed:
Release 4.2

chaunceygardiner committed:
Release v5.0

chaunceygardiner committed:
compare report was using obsolete html tag.

chaunceygardiner committed:
Remove trailing slashes from void elements (<br>, <img>, <link>)
  throughout the forecast and compare skins.  The 5.0 HTML5 doctype
  exposed them: the Nu Html Checker (check_weewx_html) flagged every
  self-closed void element ("Trailing slash on void elements has no
  effect and interacts badly with unquoted attribute values"),
  producing ~2,700 validator messages across the generated pages.
  108 slashes removed across 17 skin files; rendered output unchanged.

  Also folds in the already-staged compare-skin doctype fix (XHTML 1.0
  Strict -> HTML5) that was sitting under "Unreleased".

  Version bumped to 5.0.1 in install.py, forecast.py, and skin.conf.

chaunceygardiner committed:
Release v5.1

Honor the shutdown flag weewx 5.5 hands to report generators.
ForecastPlotGenerator checks it while scanning the forecast database
and while collecting data, and passes it to weewx's image generator,
which checks it between plots.  Stopping weewxd mid-report no longer
waits for every comparison plot to be drawn.  Older weewx releases
have no such flag and are unaffected.

Fix weewxd being unable to shut down during a forecast update under
single_thread = True: weewxd raises Terminate from its SIGTERM handler,
inside whatever the main thread is running, and do_forecast's broad
exception handler swallowed it as a forecast failure.  Verified against
a live weewxd 5.5.0: exits in 0.1s with the fix, still running 20s after
SIGTERM without it.

Fix the error path in do_forecast: a failure in get_forecast raised
UnboundLocalError from the handler itself, which reports dbm_dict --
not assigned until after the forecast is fetched -- so the real error
was lost and the exception escaped.

Fix the compare report failing whenever there is no forecast data to
plot (source not yet downloaded, misspelled, or every issue older than
issued_since).  The image generator was handed a zero or negative time
span and threw "scaletime called with tmax <= tmin" for every plot,
once per report cycle.  The generator now logs that it has nothing to
plot and stops.

Move the comparison generator's scratch database from the fixed
/var/tmp/fpg.sdb to SQLITE_ROOT, beside the station's own databases, so
two weewx instances on one machine cannot fight over one file and it no
longer sits in a sticky, world-writable directory.  A MySQL-only
station gets a pid-stamped file in the system temp directory.

Rework the comparison generator's tests: they ran against an empty
forecast database, so all of them passed while generating no plots at
all.  They now build a forecast database with real data anchored to the
current time and assert which plots come out.  Drop the empty
bin/user/test/forecast.sdb fixture, which is no longer used.  Add tests
pinning the Terminate and get_forecast error paths.
